spec/javascripts/models/beta/story_spec.js

```diff
@@ -246,107 +246,179 @@ describe('Story model', function () {
-        const newAttributes = { storyType: type };
+      describe(`when state is ${UNSTARTED}`, () => {
+        describe(`when new state is ${UNSCHEDULED}`, () => {
+          it("change estimate to ''", () => {
+            const story = { _editing: { storyType: FEATURE, state: UNSTARTED, estimate: 1 } };
+            const newAttributes = { state: UNSCHEDULED }
+    
+            const changedStory = Story.editStory(story, newAttributes);
+    
+            expect(changedStory._editing.estimate).toEqual('');
+            expect(changedStory._editing.state).toEqual(UNSCHEDULED);
+          });
+        });
+      });
 
-        const changedStory = Story.editStory(story, newAttributes);
+      estimatedValues.forEach(estimatedValue => {
+        describe(`when estimate is ${estimatedValue}`, () => {
+          describe(`when new estimate is ""`, () => {
+            it(`change state to unscheduled `, () => {
+              const story = { _editing: { storyType: FEATURE, estimate: estimatedValue, state: STARTED } };
+              const newAttributes = { estimate: '' };
+  
+              const changedStory = Story.editStory(story, newAttributes);
+  
+              expect(changedStory._editing.state).toEqual(UNSCHEDULED);
+              expect(changedStory._editing.estimate).toEqual('');
+            });
+          });
+        });
+      });
 
-        expect(changedStory).toEqual({
-          _editing: {
-            labels: [],
-            storyType: type,
-            estimate: '',
-            state: 'unstarted',
-            _isDirty: true
-          }
+      describe(`when estimate is ""`, () => {
+        estimatedValues.forEach(estimatedValue => {
+          describe(`when new estimate is ${estimatedValue}`, () => {
+            it(`change story state to unstarted`, () => {
+              const story = { _editing: { storyType: FEATURE, estimate: '', state: UNSCHEDULED } };
+              const newAttributes = { estimate: estimatedValue };
+  
+              const changedStory = Story.editStory(story, newAttributes);
+      
+              expect(changedStory._editing.state).toEqual(UNSTARTED);
+              expect(changedStory._editing.estimate).toEqual(estimatedValue);
+            });
+          });
         });
       });
-    })
+    });
 
-    it('change story estimate and sets _isDirty to true', () => {
-      const story = { _editing: { estimate: 1, storyType: 'feature', state: 'unstarted' } };
-      const newAttributes = { estimate: 2 };
+    notFeatureTypes.forEach(noFeatureType => {
+      describe(`when story type is ${noFeatureType}`, () => {
+        const otherNotFeatureTypes = notFeatureTypes.filter(type => type !== noFeatureType);
 
-      const changedStory = Story.editStory(story, newAttributes);
+        otherNotFeatureTypes.forEach(otherNotFeatureType => {
+          describe(`when new story type is ${otherNotFeatureType}`, () => {
```

Comment:
You can apply this style https://github.com/Codeminer42/cm42-central/pull/576/files#diff-e7cad8ffcafac32387e195a4957fe897R257-R273 here.